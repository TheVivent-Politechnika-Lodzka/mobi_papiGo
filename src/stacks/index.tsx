import { createNativeStackNavigator } from '@react-navigation/native-stack';
import React, { useEffect, useState } from 'react';
import auth from '@react-native-firebase/auth';
import LoginScreen from '../screens/auth/loginScreen';
import RegisterScreen from '../screens/auth/registerScreen';
import LandingScreen from '../screens/auth/landingScreen';
import MapScreen from '../screens/main/mapScreen';
import UserScreen from '../screens/main/userScreen';
import useFirestore from '../auth/useFirestore';
import InventoryScreen from '../screens/main/InventoryScreen';
import AnimalsScreen from '../screens/main/animalsScreen';

const MainStack = createNativeStackNavigator();
const AuthStack = createNativeStackNavigator();

export default function AllStack() {
  useFirestore();
  const [loggedIn, setLoggedIn] = useState(false);

  useEffect(() => {
    checkLogin();
  }, []);

  const checkLogin = () => {
    auth().onAuthStateChanged((user) => {
      if (user) {
        setLoggedIn(true);
      } else {
        setLoggedIn(false);
      }
    });
  };

  if (loggedIn) {
    return (
      <MainStack.Navigator
        screenOptions={{ headerShown: false }}
        initialRouteName="Map"
      >
        <MainStack.Screen name="Map" component={MapScreen} />
        <MainStack.Screen name="User" component={UserScreen} />
        <MainStack.Screen name="Inventory" component={InventoryScreen} />
        <MainStack.Screen name="Animals" component={AnimalsScreen} />
      </MainStack.Navigator>
    );
  }

  // not logged in
  return (
    <AuthStack.Navigator
      screenOptions={{ headerShown: false }}
      initialRouteName="Landing"
    >
      <AuthStack.Screen name="Landing" component={LandingScreen} />
      <AuthStack.Screen name="Login" component={LoginScreen} />
      <AuthStack.Screen name="Register" component={RegisterScreen} />
    </AuthStack.Navigator>
  );
}
