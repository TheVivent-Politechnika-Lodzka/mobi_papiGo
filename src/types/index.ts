import { v4 as uuid } from 'uuid';

export class DbData {
  id: string = '';
  registrationDate: Date = new Date();
  items: Item[] = [];
  animals: Animal[] = [];
}

export class Item {
  id: string = uuid();
  name: string = '';
  stars: number = 0;
  buff: Buff = new Buff();
}

export class Animal {
  id: string = uuid();
  name: string = '';
  type: AnimalType = null;
  stars: number = 0;
  stats: Stats = new Stats();
  item: Item | null = null;
}

export class Buff {
  energy?: number = 0;
  strength?: number = 0;
  agility?: number = 0;
  range?: number = 0;
}

export class Stats extends Buff {
  energy: number = 0;
  strength: number = 0;
  agility: number = 0;
  range: number = 0;
}

export type AnimalType = null | 'cat' | 'dog';
