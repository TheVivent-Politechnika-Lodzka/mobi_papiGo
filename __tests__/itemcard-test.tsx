import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { Item } from '../src/types';
import ItemCard from '../src/components/item-card';



it('Card test', () => {
  const item = new Item();
  const button = renderer.create(<ItemCard item={item}/>).toJSON();
  expect(button).toMatchSnapshot();
});