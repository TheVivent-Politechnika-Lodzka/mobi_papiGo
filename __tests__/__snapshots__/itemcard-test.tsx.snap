// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F6EABE",
      "borderColor": "#e1e8ee",
      "borderRadius": 10,
      "borderWidth": 1,
      "flex": 1,
      "height": 350,
      "margin": "10%",
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
      "width": "80%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#43484d",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
      testID="cardTitle"
    >
      
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#bcbbc1",
          "borderBottomWidth": 0.5,
          "marginBottom": 15,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "opacity": 1,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Object {
            "testUri": "../../../assets/collar.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 150,
            "left": 0,
            "marginBottom": "3%",
            "position": "absolute",
            "resizeMode": "contain",
            "right": 0,
            "top": 0,
            "width": null,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        collapsable={false}
        importantForAccessibility="no-hide-descendants"
        nativeID="animatedComponent"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 150,
              "justifyContent": "center",
              "marginBottom": "3%",
              "resizeMode": "contain",
              "width": null,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "height": 150,
            "marginBottom": "3%",
            "resizeMode": "contain",
            "width": null,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#bcbbc1",
          "borderBottomWidth": 0.5,
          "marginBottom": 15,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": "2%",
            "marginVertical": "2%",
            "width": "31%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#C8E3D4",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": "8%",
              "width": "90%",
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 20,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome5Free-Solid",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {
                        "fontWeight": "900",
                      },
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "marginLeft": "3%",
              }
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": "2%",
            "marginVertical": "2%",
            "width": "31%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#C8E3D4",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": "8%",
              "width": "90%",
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 20,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome5Free-Solid",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {
                        "fontWeight": "900",
                      },
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "marginLeft": "3%",
              }
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": "2%",
            "marginVertical": "2%",
            "width": "31%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#C8E3D4",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": "8%",
              "width": "90%",
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 20,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome5Free-Solid",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {
                        "fontWeight": "900",
                      },
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "marginLeft": "3%",
              }
            }
          >
            0
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginHorizontal": "25%",
            "marginVertical": "2%",
            "width": "50%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#C8E3D4",
              "borderRadius": 10,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": "8%",
              "width": "90%",
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 20,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "marginLeft": "3%",
              }
            }
          >
            0
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
